src/java.base/share/classes/java/util/HexDigits.java

```diff
@@ -105,14 +105,14 @@ private HexDigits() {
     /**
      * Combine two hex shorts into one int based on big endian
      */
-    static int digit(int b0, int b1) {
+    static int packDigits(int b0, int b1) {
```

Comment:
I recommend such a refactor:
```java
/**
 * Return a big-endian packed integer for the 4 ASCII bytes for an input unsigned short value.
 */
static int packDigits16(int b) {
    return (DIGITS[(b >> 8) & 0xff] << 16) | DIGITS[b & 0xff];
}
```
And all your usages can be `HexDigits.packDigits16(((int) msb) >> 16)` and `HexDigits.packDigits16((int) msb)` without the `>> 24` and `>> 8`

Reply:
but it's return an int value

Reply:
The current code snippet still returns a int value... I think I can probably make a patch and demonstrate.